### ISSUE TITLE
Unable to run repo:create from non-git folder

### ISSUE THREAD

**Issue by sstok** [gushphp/gush#488]
```
$ gh help repo:create


 [ERROR] Unable to get the repository information, Git remote "origin" should be set for automatic detection.
```

**Comment by sstok** [comment]
OK, this requires a bit more work :sob: the RepositoryCreateCommand depends on the adapter but can't require that the command is executed from a Git repository.

We can split-up the `GitRepoSubscriber` into a listener for adapter-aware commands and a second listener for commands that need the org/repo options. Plus as a small bonus we are further reducing the complexity of `GitRepoSubscriber::initialize()`

Also using having the `--org` option in this command doesn't make sense, I actually assumed this was 'used' as target-org :smile:

**Comment by cordoval** [comment]
hmm not sure what is the use case here, to bootstrap a project on an empty folder?

**Comment by sstok** [comment]
Yes :) I have initialized an empty git repository and want to create an empty repository on GitHub.

**Comment by cordoval** [comment]
I assume you mean repo:create for creating a repository on upstream github or other? I would look more into keeping this even more simpler and have a global configuration that does not depend on local folder. However the splitting into weird class names is not fun, if we do we have to make code more plain and using better names and document the reason why we do that.

**Comment by sstok** [comment]
Closed in favor of #459
